src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -476,40 +498,19 @@ RebuildOnConflict(Oid relationId, OnConflictExpr *originalOnConflict)
 
 
 /*
- * DistributedModifyJob creates a Job for the specified query to execute the
- * provided modification task. Modification task placements are produced using
- * the "first-replica" algorithm, except modifications run against all matching
- * placements rather than just the first successful one.
- */
-Job *
-DistributedModifyJob(Query *query, Task *modifyTask)
-{
-	Job *modifyJob = NULL;
-	List *taskList = FirstReplicaAssignTaskList(list_make1(modifyTask));
-
-	modifyJob = CitusMakeNode(Job);
-	modifyJob->dependedJobList = NIL;
-	modifyJob->jobId = INVALID_JOB_ID;
-	modifyJob->subqueryPushdown = false;
-	modifyJob->jobQuery = query;
-	modifyJob->taskList = taskList;
-
-	return modifyJob;
-}
-
-
-/*
- * DistributedModifyShardInterval determines the single shard targeted by a
- * provided distributed modification command. If no matching shards exist, or
- * if the modification targets more than one one shard, this function raises
- * an error.
+ * DetermineTargetShardInterval determines the single shard targeted by a
+ * provided command. If no matching shards exist, or if the modification targets
+ * more than one one shard, this function raises an error.
  */
 static ShardInterval *
-DistributedModifyShardInterval(Query *query)
+DetermineTargetShardInterval(Query *query, TaskType taskType)
```

Comment:
Minor: What about `DetermineTargetShardInterval` to `TargetShardInterval`. Also, we don't need to pass `task_type`, since we can get it from the `query`.